src/helpers/firebaseHelpers.js

```diff
@@ -2,9 +2,25 @@
 
 export const parseData = (querySnapshot) =>
   querySnapshot.docs.reduce((acc, doc) => {
-    const { productName, listToken, timeFrame, lastPurchaseDate } = doc.data();
+    const {
+      productName,
+      timeFrame,
+      lastPurchaseDate,
+      daysUntilNextPurchase,
+      numberOfPurchases,
+      createdAt,
+    } = doc.data();
+
     return [
       ...acc,
-      { id: doc.id, productName, listToken, timeFrame, lastPurchaseDate },
+      {
+        id: doc.id,
+        productName,
+        timeFrame,
+        lastPurchaseDate,
+        daysUntilNextPurchase,
+        numberOfPurchases,
+        createdAt,
```

Comment:
I believe it's a good opportunity to use [spread operator/object literals](https://nitayneeman.com/posts/object-rest-and-spread-properties-in-ecmascript-2018/).

```js
const {id} = doc;
const product = doc.data();

return [
      ...acc,
      {
        id, ...product 
      },
    ];
```

What do you think about it?

Reply:
Yeah, much better this way. I didn't have this in mind... Just imagine retrieving much more data from Firestore and _having_ declare it every time new data comes